File: app/src/main/java/ie/dublinbuspal/android/view/favourite/FavouritesView.java
package ie.dublinbuspal.android.view.favourite;

import com.hannesdorfmann.mosby3.mvp.MvpView;

import java.util.List;

import ie.dublinbuspal.model.favourite.FavouriteStop;

public interface FavouritesView extends MvpView {

    void showFavourites(List<FavouriteStop> favouriteBusStops);

    void launchRealTimeActivity(String stopId);

    void hideProgress();

    void showError(int stringId);

}

File: app/src/main/kotlin/ie/dublinbuspal/view/favourites/FavouritesView.kt
package ie.dublinbuspal.view.favourites

import com.hannesdorfmann.mosby3.mvp.MvpView
import ie.dublinbuspal.model.favourite.FavouriteStop
import ie.dublinbuspal.model.livedata.Destination
import ie.dublinbuspal.model.livedata.LiveData

interface FavouritesView : MvpView {

    fun showFavourites(favourites: List<FavouriteStop>)

    fun showLiveData(favourites: List<FavouriteStop>, favouriteId: String, livedata: Map<Pair<String, Destination>, List<LiveData>>)

}
